### ISSUE TITLE
Replace all apiset references with classic dll references

### ISSUE THREAD

**Issue by karelz** [dotnet/corefx#14728]
[2 failures](https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/1688/) in PR https://github.com/dotnet/corefx/pull/14726#issuecomment-269396959:

**[System.IO.Pipes.Tests.NamedPipeTest_Specific.Windows_Get_NumberOfServerInstances_Succeed](https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/1688/testReport/junit/System.IO.Pipes.Tests/NamedPipeTest_Specific/Windows_Get_NumberOfServerInstances_Succeed/):**

<details>
MESSAGE:
System.BadImageFormatException : An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)
+++++++++++++++++++
STACK TRACE:
at System.IO.Pipes.Tests.Interop.GetNamedPipeHandleState(SafePipeHandle hNamedPipe, IntPtr lpState, Int32& lpCurInstances, IntPtr lpMaxCollectionCount, IntPtr lpCollectDataTimeout, StringBuilder lpUserName, Int32 nMaxUserNameSize) at System.IO.Pipes.Tests.Interop.TryGetNumberOfServerInstances(SafePipeHandle handle, Int32& numberOfServerInstances) in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.Pipes\tests\Interop.cs:line 58 at System.IO.Pipes.Tests.NamedPipeTest_Specific.<Windows_Get_NumberOfServerInstances_Succeed>d__5.MoveNext() in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.Pipes\tests\NamedPipeTests\NamedPipeTest.Specific.cs:line 185 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
</details>

<br/>

**[System.IO.Pipes.Tests.NamedPipeTest_Specific.Windows_GetImpersonationUserName_Succeed](https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/1688/testReport/junit/System.IO.Pipes.Tests/NamedPipeTest_Specific/Windows_GetImpersonationUserName_Succeed/):**

<details>
```
MESSAGE:
System.BadImageFormatException : An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)
+++++++++++++++++++
STACK TRACE:
at System.IO.Pipes.Tests.Interop.GetNamedPipeHandleState(SafePipeHandle hNamedPipe, IntPtr lpState, Int32& lpCurInstances, IntPtr lpMaxCollectionCount, IntPtr lpCollectDataTimeout, StringBuilder lpUserName, Int32 nMaxUserNameSize) at System.IO.Pipes.Tests.Interop.TryGetImpersonationUserName(SafePipeHandle handle, String& userName) in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.Pipes\tests\Interop.cs:line 44 at System.IO.Pipes.Tests.NamedPipeTest_Specific.<Windows_GetImpersonationUserName_Succeed>d__6.MoveNext() in D:\j\workspace\windows_nt_de---4526f5ff\src\System.IO.Pipes\tests\NamedPipeTests\NamedPipeTest.Specific.cs:line 207 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```
</details>

**Comment by jkotas** [comment]
The crash was caused by tests still referencing API sets. We have dropped API sets recently, and this PR was the first one that brought the API sets over.

I see there are number of other API set references in tests. All of them should be fixed.

**Comment by jkotas** [comment]
cc @JohnChen0

**Comment by jkotas** [comment]
Keep open to fix the other API set references in tests

**Comment by danmosemsft** [comment]
@JohnChen0 do you plan to update these also? There seem to be 34.

**Comment by xoofx** [comment]
Not sure it was already tracked it, but it seems that native dll like `clrcompression.dll` still imports `api-ms-win-crt-runtime-l1-1-0.dll` and `api-ms-win-crt-heap-l1-1-0.dll`

**Comment by danmosemsft** [comment]
This is causing a bunch of test failures on Windows 7. @JohnChen0 do you plan to fix these or should I ask someone to do so? Also can you comment on @xoofx's note above?

```
System.Diagnostics.Process.Tests	System.Diagnostics.Tests.ProcessTests	TestMaxWorkingSet
		TestMinWorkingSet
System.Dynamic.Runtime.Tests	ManagedTests.DynamicCSharp.Conformance.dynamic.context.property.autoproperty.regclass.regclass007…	DynamicCSharpRunTest
System.Globalization.Tests	System.Globalization.Tests.CultureInfoAll	TestAllCultures
System.IO.FileSystem.Watcher.Tests	System.IO.Tests.Directory_NotifyFilter_Tests	FileSystemWatcher_Directory_NotifyFilter_Security
	System.IO.Tests.File_NotifyFilter_Tests	FileSystemWatcher_File_NotifyFilter_Security
```

**Comment by danmosemsft** [comment]
You can see what dll they are trying to load by clicking through https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170104.02 eg to https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170104.02/workItem/System.IO.FileSystem.Watcher.Tests/analysis/xunit/System.IO.Tests.Directory_NotifyFilter_Tests~2FFileSystemWatcher_Directory_NotifyFilter_Security(filter:%20Attributes)

**Comment by JohnChen0** [comment]
@danmosemsft I'll look into fixing the usage of API Set in test code.

@xoofx `api-ms-win-crt-runtime-l1-1-0.dll` and `api-ms-win-crt-heap-l1-1-0.dll` are still the official DLL names to use to dynamically link against C runtime. These DLL names came from `ucrt.lib` used by Visual Studio 2015. We could statically link to the C++ runtime to avoid these imports, but I think that would be separate feature, independent of current API Set work.

**Comment by karelz** [comment]
@xoofx if you think the CRT part is useful, we may de-dupe you original bug you filed. To assess priority, I'd like to know what is the value of such work and what is the downside if we do not do it.

**Comment by xoofx** [comment]
@karelz This is typically to avoid issues like ["dll missing from your computer"](https://github.com/dotnet/cli/issues/5201) where you need to install the VC++ redistributable runtime in order to use a self contained dotnetcore app. A .NET runtime like coreclr+corefx should not have any dependencies to a C++ dll runtime. With all the work being done already for apiset, it would be a pity to just leave one or two dlls behind with these dependencies...

**Comment by karelz** [comment]
@xoofx in my mind API sets are different than CRT dependency. Can you please either reopen your original bug and scope it to CRT or open a new one? We can have detailed discussion there ...

**Comment by xoofx** [comment]
@karelz created #14865

**Comment by JohnChen0** [comment]
Resolved with PR #14876.
